### ISSUE TITLE
[WIP] API Versioning

### ISSUE THREAD

**Issue by Ener-Getick** [EXSyst/Api#3]
I am thinking about supporting the api versioning.
The major problem is that there is no common standard...

The most used solutions are:
- a versioned URI (``/v1/users``)
- a query parameter containing a version number (``/users?version=v1``)
- a query parameter containing a date (``/users?version=2011-10-01``)
- a custom media-type in the ``Accept`` header (``Accept: application/json.v1``)
- a custom header ``X-Accept-Version`` containing a constraint (see  https://getcomposer.org/doc/articles/versions.md) (``X-Accept-Version: ~1.0``)

So if this package support the API versioning it has to support most of this solutions. One class calling other classes that check if they can get the current version must be a good beginning.
If we set a ``Request`` attribute, it would even be possible to use the routing requirements.

**Comment by Ener-Getick** [comment]
I think about using a pre-configured list of versions:
```yaml
exsyst_api:
    versioning:
        versions:
            v1: { release: 2011-10-12 }
            v2: ~
        resolvers:
            uri: true
            query: true
            date: true
            header: true
            constraint: true
```

**Comment by Ener-Getick** [comment]
Fixed by #4
